modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitionsModals.tsx

```diff
@@ -236,6 +236,29 @@ export function ViewObjectDefinitionsModals({
 				/>
 			)}
 
+			{showModal.objectDefinitionOnRootModelDeletionNotAllowed &&
+				selectedObjectDefinition &&
+				Liferay.FeatureFlags['LPS-187142'] && (
+					<ModalObjectFieldDeletionNotAllowed
+						content={
+							<span
+								dangerouslySetInnerHTML={{
+									__html: Liferay.Language.get(
+										'to-delete-this-object-you-must-first-disable-inheritance-and-delete-its-relationships'
+									),
+								}}
+							/>
+						}
+						onVisibilityChange={() =>
```

Comment:
Again, existing pattern, but this is simply an onClose from Clay's `useModal`, Why give it another name? If is was called both on open and on close, ok, but is passed directly to onClose. It looks misleading.